### ISSUE TITLE
Xonsh interprets certain nested structures incorrectly

### ISSUE THREAD

**Issue by wrywerytwreywery** [scopatz/xonsh#85]
The following do not work as expected in xonsh:

```
$ [[i] for i in range(5)]
[0, 1, 2, 3, 4]
```
```
$ [(i,) for i in range(5)]
[0, 1, 2, 3, 4]
```
```
$ [[1],[2],[3],[4],]
[1, [2], [3], [4]]
```

I am not sure exactly what the pattern is, but I did notice these failure modes and figured it might be helpful to point them out.  I know there had been some issues in the past with lists inside of dictionaries; maybe this is related?

**Comment by scopatz** [comment]
Thanks for reporting @wrywerytwreywery! This is clearly a parser bug.  I think I know where too look for it.  I'll take a stab tomorrow, because I am too tired tonight.

**Comment by scopatz** [comment]
#100 should address this @wrywerytwreywery

**Comment by wrywerytwreywery** [comment]
Fixed with #100.  Closing.
